### PR TITLE
added login funcionality

### PR DESCRIPTION
**Pull request by soniaigartua**
Ready for opinions .... I added the login funcionality, and first design of registration funcionality.

### CODE REVIEW

**Comment by soniaigartua** [comment]
Evidences:
loginScreen
Test:
Anotación 2019-07-30 114158 testLogin

File: app/src/main/java/com/example/pps_tudai/mvp/presenter/LoginPresenter.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                loginView.showOnAuthenticationErrorMessage();
+            }
+        }
+        else{
+            loginView.showOnEmailErrorMessage();
+        }
+    }
+
+    private boolean isvalid(String email) {
+        String regex = "^[\\w-_\\.+]*[\\w-_\\.]\\@([\\w]+\\.)+[\\w]+[\\w]$";
```
Comment (SartiMau) [inline app/src/main/java/com/example/pps_tudai/mvp/presenter/LoginPresenter.java]: Put this regex in a constant pls. 
Maybe you can create a StringUtils class and put there all the constants, so you will be able to re-use from other places.

File: app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import butterknife.ButterKnife;
+
+import static androidx.core.content.ContextCompat.startActivity;
+
+public class LoginView {
+
+    @BindView(R.id.et_email) EditText email;
+    @BindView(R.id.et_password) EditText password;
+    @BindView(R.id.btn_enter_login) Button btnLogin;
+    @BindView(R.id.btn_cancel_login) Button btnCancel;
```
Comment (SartiMau) [inline app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java]: Do you need the Buttons here? if not, please delete them.

File: app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+
+    public String getPassword() {
+        return password.getText().toString();
+    }
+
+    public void showOnAuthenticationErrorMessage() {
+        if (activityWeak.get() != null) {
+            Toast.makeText(activityWeak.get(), "Password is incorrect", Toast.LENGTH_SHORT).show();
```
Comment (SartiMau) [inline app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java]: no hardoce pls.. You can use string resources from here.

File: app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        }
+    }
+
+    public void showOnEmailErrorMessage() {
+        if (activityWeak.get() != null) {
+            Toast.makeText(activityWeak.get(), "Email address is incorrect", Toast.LENGTH_SHORT).show();
```
Comment (SartiMau) [inline app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java]: same

File: app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        }
+    }
+
+    public void showWelcomeScreen() {
+        if (activityWeak.get() != null) {
+            Toast.makeText(activityWeak.get(), "Welcome to lorenaApp", Toast.LENGTH_SHORT).show();
```
Comment (SartiMau) [inline app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java]: same

File: app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        }
+    }
+
+    public void showDataEmptyScreen() {
+        if (activityWeak.get() != null) {
+            Toast.makeText(activityWeak.get(), "Email Address or Password can not be empty", Toast.LENGTH_SHORT).show();
```
Comment (SartiMau) [inline app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java]: same

File: app/src/main/java/com/example/pps_tudai/mvp/view/RegistrationView.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+
+    public String getPassword() {
+        return password.getText().toString();
+    }
+
+    public void showRegistrationScreenOK() {
+        if (activityWeak.get() != null) {
+            Toast.makeText(activityWeak.get(), "Registration complete", Toast.LENGTH_SHORT).show();
```
Comment (SartiMau) [inline app/src/main/java/com/example/pps_tudai/mvp/view/RegistrationView.java]: same

File: app/src/main/java/com/example/pps_tudai/mvp/view/RegistrationView.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+        }
+    }
+
+    public void showDataEmptyScreen() {
+        if (activityWeak.get() != null) {
+            Toast.makeText(activityWeak.get(), "Can not have empty fields", Toast.LENGTH_SHORT).show();
```
Comment (SartiMau) [inline app/src/main/java/com/example/pps_tudai/mvp/view/RegistrationView.java]: same

File: app/src/test/java/com/example/pps_tudai/mvp/presenter/LoginPresenterTest.java [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        loginModel = new LoginModel();
+        loginPresenter = new LoginPresenter(loginModel,loginView);
+
+    }
+
+
+//    @Test
```
Comment (SartiMau) [inline app/src/test/java/com/example/pps_tudai/mvp/presenter/LoginPresenterTest.java]: remove commented code pls

File: app/src/test/java/com/example/pps_tudai/mvp/presenter/LoginPresenterTest.java [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+        when(loginView.getUserEmail()).thenReturn("lorena.example.com");
+        when(loginView.getPassword()).thenReturn("123456");
+        loginPresenter.onLoginPressed();
+
+        verify(loginView).showOnEmailErrorMessage();
+    }
+
+    @Test
+    public void loginEmptyData() {
+        when(loginView.getUserEmail()).thenReturn("");
```
Comment (SartiMau) [inline app/src/test/java/com/example/pps_tudai/mvp/presenter/LoginPresenterTest.java]: no hardcode pls.. use constant, you can put it on the StringUtils class

File: app/src/test/java/com/example/pps_tudai/mvp/presenter/LoginPresenterTest.java (hunk 9 above)
Comment (soniaigartua) [reply app/src/test/java/com/example/pps_tudai/mvp/presenter/LoginPresenterTest.java]: done!

File: app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java (hunk 3 above)
Comment (soniaigartua) [reply app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java]: done!

File: app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java (hunk 4 above)
Comment (soniaigartua) [reply app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java]: done!

File: app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java (hunk 5 above)
Comment (soniaigartua) [reply app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java]: done!

File: app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java (hunk 6 above)
Comment (soniaigartua) [reply app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java]: done!

File: app/src/main/java/com/example/pps_tudai/mvp/view/RegistrationView.java (hunk 7 above)
Comment (soniaigartua) [reply app/src/main/java/com/example/pps_tudai/mvp/view/RegistrationView.java]: done!

File: app/src/main/java/com/example/pps_tudai/mvp/view/RegistrationView.java (hunk 8 above)
Comment (soniaigartua) [reply app/src/main/java/com/example/pps_tudai/mvp/view/RegistrationView.java]: done!

File: app/src/test/java/com/example/pps_tudai/mvp/presenter/LoginPresenterTest.java (hunk 10 above)
Comment (soniaigartua) [reply app/src/test/java/com/example/pps_tudai/mvp/presenter/LoginPresenterTest.java]: DONE!!

File: app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java (hunk 2 above)
Comment (soniaigartua) [reply app/src/main/java/com/example/pps_tudai/mvp/view/LoginView.java]: DONE!

File: app/src/main/java/com/example/pps_tudai/mvp/presenter/LoginPresenter.java (hunk 1 above)
Comment (soniaigartua) [reply app/src/main/java/com/example/pps_tudai/mvp/presenter/LoginPresenter.java]: done!

File: app/src/main/java/com/example/pps_tudai/mvp/view/RegistrationView.java [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+    public String getUserSurname() {
+        return surname.getText().toString();
+    }
+
+    public String getUserEmail() {
+        return email.getText().toString();
+    }
+
+
```
Comment (lucianoeli) [inline app/src/main/java/com/example/pps_tudai/mvp/view/RegistrationView.java]: remove empty line please

File: app/src/test/java/com/example/pps_tudai/mvp/presenter/LoginPresenterTest.java (hunk 10 above)
Comment (SartiMau) [reply app/src/test/java/com/example/pps_tudai/mvp/presenter/LoginPresenterTest.java]: "" still here

File: app/src/main/java/com/example/pps_tudai/mvp/presenter/LoginPresenter.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+                loginView.showOnAuthenticationErrorMessage();
+            }
+        }
+        else{
+            loginView.showOnEmailErrorMessage();
+        }
+    }
+
+    private boolean isvalid(String email) {
+        return email.matches(MODELO_EMAIL_VALIDO);
```
Comment (lucianoeli) [inline app/src/main/java/com/example/pps_tudai/mvp/presenter/LoginPresenter.java]: names should be in english

File: app/src/main/res/layout/activity_registration.xml [hunk 13]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    android:gravity="center">
+    tools:context=".RegistrationActivity">
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_registration.xml]: this is not fine.. please delete it

File: app/src/main/res/layout/activity_login.xml [hunk 14]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    android:gravity="center">
+    tools:context=".LoginActivity">
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_login.xml]: this is not fine.. please delete it

File: app/src/test/java/com/example/pps_tudai/mvp/presenter/LoginPresenterTest.java (hunk 10 above)
Comment (soniaigartua) [reply app/src/test/java/com/example/pps_tudai/mvp/presenter/LoginPresenterTest.java]: DONE!!

File: app/src/main/java/com/example/pps_tudai/mvp/view/RegistrationView.java (hunk 11 above)
Comment (soniaigartua) [reply app/src/main/java/com/example/pps_tudai/mvp/view/RegistrationView.java]: DONE!

File: app/src/main/java/com/example/pps_tudai/mvp/presenter/LoginPresenter.java (hunk 12 above)
Comment (soniaigartua) [reply app/src/main/java/com/example/pps_tudai/mvp/presenter/LoginPresenter.java]: DONE!

File: app/src/main/res/layout/activity_registration.xml (hunk 13 above)
Comment (soniaigartua) [reply app/src/main/res/layout/activity_registration.xml]: I did not delete this comand .... I wrote it correctly inside the tag

File: app/src/main/res/layout/activity_login.xml (hunk 14 above)
Comment (soniaigartua) [reply app/src/main/res/layout/activity_login.xml]: I did not delete this comand .... I wrote it correctly inside the tag

**Pull request merged**